google/ggrc-core
# GGRC-4718/GGRC-4719/GGRC-4721/GGRC-4722/GGRC-5201 Object Review with email notification

# Dependencies

This PR contains only FE part and is `on hold` until the BE dependencies are merged:

- [ ] #8396 

# Issue description

New Object Review functionality - part 1. The list of objects with object review are presented in the GRC Bible doc (Control, Program, Regulation, Objective, Clause, Contract, Policy, Risk, Standard, Threat, Requirement). Old review functionality was connected with workflows and cycles, but now it will be connected with the Object itself.

# Steps to test the changes
**Log in as user with Edit rights (admin or global creator, editor)**
1. Create not scoping object, f.e. Control. By default it has Unreviewed state. Also there should be Mark Reviewed and Request Review buttons
2. Click Request Review - you will see pop up. Add user with READ permissions to reviewers list and save.
3. Added user should be displayed in reviewers list on the Control page.
4. Click Mark Reviewed. Control state should be changed to Reviewed and you will see info about last review (user and time). You (current user) should be added to the reviewers list. Also there should be notification this Undo link at the page bottom.
5. Undo link should change state to Unreviewed.
6. Review Control again and check that after changing attribute (f.e. description) or map/unmap other Object Control has Unreviewed state.
7. If object in Unreviewed state - Review History is displayed and on click you navigate to Change log tab with active filter. Filter allows us to see the changes after the last changing to Reviewed state.
8. Open any Scope object - it should not have Object Review block.

**Log in as user with READ rights**
1. Open Control from part 1. You should see buttons and be able to review Control.
2. Open another object where you are not a reviewer. You will not see the buttons - only state.

# Solution description

Create separate Review model in service-models. Create component for Object Review block on the page. So it should display state, info about last review and buttons. Component for Request Review popup is using Simple Modal. Some helpful functions are moved to the object-review-utils.
Also there were removed all components and templates for old object review.
Scope Objects should not have review, so was decided to add mustache helper which checks is it scope model or not and use it in base_objects/info.mustache.

# Sanity checklist




# PR Review checklist
